src/util/stop.c

```diff
@@ -74,6 +104,7 @@ static double sc(double x, double smin, double smax)
 int nlopt_stop_xs(const nlopt_stopping * s, const double *xs, const double *oldxs, const double *scale_min, const double *scale_max)
 {
     unsigned i;
+    /* TODO: scale ‖Δx‖... but not with w */
```

Comment:
Yeah, it seems like we'd need another variant of `vector_norm` and `diff_norm` that takes `scale_min` and `scale_max` arguments.  Or simply add these parameters and pass `NULL` in cases where they aren't used.